src/transformers/generation/utils.py

```diff
@@ -3068,14 +3066,14 @@ def beam_sample(
                 next_token_logits, dim=-1
             )  # (batch_size * num_beams, vocab_size)
 
-            next_token_scores_processed = logits_processor(input_ids, next_token_scores)
-            next_token_scores = next_token_scores_processed + beam_scores[:, None].expand_as(next_token_scores)
+            next_token_scores = logits_processor(input_ids, next_token_scores)
             next_token_scores = logits_warper(input_ids, next_token_scores)
+            next_scores = next_token_scores + beam_scores[:, None].expand_as(next_token_scores)
```

Comment:
@patrickvonplaten ⚠️ important detail here: `beam_sample` was applying `logits_warper` on the accumulated score, as opposed to the token score. 

Since the logits warpers are used to adapt the token-level log probs, I am assuming this is bug -- hence this change :)

Reply:
This was actually intentional, we never changed it simply because it's backward breaking and because it's not clear that `logits_warper` should be applied **before** adding the beam_scores. 

Note that `logits_warper` is not the same as `logits_processor`. Logits warper includes top_p, top_k etc... those filters are supposed to be applied directly **before** the softmax . Note that top_k = 50 means that softmax can only pick from 50 values from the logits tensor (see blog post here: https://huggingface.co/blog/how-to-generate#top-k-sampling).

=> So I'm not in favor of this change. It's an exotic case as beam sample is a semi-official generation method, but IMO it's better to apply the logits warper before the softmax because all its parameters (temp, top_p, top_k) are tailored to be applied directly before the softmax. Also it's breaking if we change it here

Reply:
This was previously already discussed here as a reference: https://github.com/huggingface/transformers/pull/5420#discussion_r449779867